test-tools/README.md

```diff
@@ -24,3 +24,136 @@ when tests are run in a Docker container (during CI) running as root.
 
 **NOTE:** To limit changes to the files obtained from the JUnit code base to the absolute minimum, 
 the Terracotta copyright header is intentionally omitted and SpotBugs/FindBugs is intentionally suppressed.
+
+
+## Dependency Declarations for `terracotta-utilities-test-tools`
+
+The dependency declarations needed for the `terracotta-utilities-test-tools` module depends on:
+
+1. the build system, e.g. Maven vs Gradle
+2. the tools needed
+
+The `terracotta-utilities-test-tools` module contains the following general tool categories:
+
+1. tools used for writing JUnit4/Hamcrest tests
+2. tools used to verify Logback event output
+3. tools not requiring either JUnit4/Hamcrest or Logback
+
+The dependency declarations are described below.
+
+### Maven Declarations
+
+For Maven builds, the `terracotta-utilities-test-tools` module has several _optional_ dependencies.
+The dependency declarations needed vary with the tools you wish to use.
+
+#### Minimal Declaration
+
+The following Maven dependency declaration is required for all uses of `terracotta-utilties-test-tools`:
+
+    <dependency>
+      <groupId>org.terracotta</groupId>
+      <artifactId>terracotta-utilities-test-tools</artifactId>
+      <version>${toolVersion}</version>
+      <scope>test</scope>
+    </dependency>
+
+The above declaration supports the use of the tooling not requiring JUnit 4, Hamcrest, or Logback,
+for example, `Diagnostics`.  To use the JUnit4/Hamcrest and/or Logback tools, additional dependencies
+must be added.
+
+#### JUnit4/Hamcrest Tooling
+
+To use the JUnit4/Hamcrest tools, the following declarations must be added:
+
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>4.12</version>
+      <scope>test</scope>
+      <exclusions>
```

Comment:
This exclusion isn't needed unless you enforce dependency convergence, right?

Reply:
Well ... JUnit drags in Hamcrest 1.3 and test-tools wants Hamcrest 2.2.  Especially since Hamcrest packaging changed between 1.3 and 2.2, it's definitely _cleanest_ to exclude the JUnit-pulled version of Hamcrest and specify the desired version of Hamcrest.  IIRC, if you don't exclude Hamcrest, JUnit still pulls in `hamcrest-core:1.3` -- the Hamcrest documentation about up-versioning wants a declaration for `hamcrest-library` to replace the old `hamcrest-core` dependency.  According to the documentation, _order_ is an issue.  Exclusion is cleaner and not order-dependent.